spec/system/planning_applications/constraints_spec.rb

```diff
@@ -0,0 +1,116 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+RSpec.describe "Planning Application Assessment", type: :system do
+  let!(:assessor) { create :user, :assessor, local_authority: @default_local_authority }
+
+  let!(:planning_application) do
+    create :planning_application, :not_started, local_authority: @default_local_authority
+  end
+
+  before do
+    sign_in assessor
+    visit planning_application_path(planning_application)
+    click_button "Constraints"
+  end
+
+  context "Editing constraints" do
+    it "can add and change constraints" do
+      click_link("Update")
+
+      check("Listed building")
+      check("Broads")
+      check("Flood zone 3")
+      uncheck("Conservation Area")
+      # The factory provides Conservation Areaa as a pre-existing constraint, so checking it will remove it
```

Comment:
"edit" / "update" are much easier to read / understand when declaring what we are updating at the top - factories should make it easier to create default attributes but we need to know what those attributes are in order to make sure the update is working correctly - so I'd say it needs to be part of this file. Could we declare the list of attributes at the top that we are modifying here, otherwise we need to check in 2 different places (the factory and here) to know what was updated

Reply:
Agreed - have updated this with the constraints explicitly set in the spec 👍